server/methods/core/shop.js

```diff
@@ -81,17 +80,24 @@ Meteor.methods({
     }
 
     // we'll accept a shop object, or clone the current shop
-    const seedShop = shopData || Collections.Shops.findOne(Reaction.getPrimaryShopId());
+    let seedShop = shopData;
+
+    if (_.isEmpty(seedShop)) {
+      const count = Collections.Shops.find().count() || "";
+      seedShop = Reaction.getPrimaryShop();
+      // generate unique shop name
+      seedShop.name = seedShop.name + count;
+    }
+
+    // ensure unique id
+    // (shouldn't Mongo handle this for us? with lesser risk of collision?)
+    seedShop._id = Random.id();
```

Comment:
Note that I have some significant changes to this in https://github.com/reactioncommerce/reaction/pull/3331, which most likely will be merged before this PR and require some conflict resolution. My changes include letting Mongo/Meteor generate the `_id` as you mention.

https://github.com/reactioncommerce/reaction/pull/3331/files#diff-a85b4a331f34703f51a72e9728869fe2